services/meta/client.go

```diff
@@ -100,6 +100,12 @@ func (c *Client) Close() error {
 	return nil
 }
 
+func (c *Client) Data() *Data {
```

Comment:
Just name it something really horrible (`internalDataStateObject`) and cc Ben for code review.  He'll suggest a good name.  :)

`cache`, `cachedData`, `dataCache`, `d`?